zerver/tests/test_templates.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+
+from django.test import TestCase
+from django.template.loader import get_template
+
+from zerver.models import get_user_profile_by_email
+from zerver.lib.test_helpers import get_all_templates
+
+
+class get_form_value(object):
+    def __init__(self, value):
+        self._value = value
+
+    def value(self):
+        return self._value
+
+
+class DummyForm(dict):
+    pass
+
+
+class TemplateTestCase(TestCase):
+    """
+    Tests for checking the rendering of backend templates.
+    """
+
+    def test_templates(self):
+        email = "hamlet@zulip.com"
+        user_profile = get_user_profile_by_email(email)
+
+        # `shallow_tested` is used to tell the receiver that the
+        # template was not rendered in an actual logical test.
+        # Note: `context` just holds dummy values to make the test
+        # pass. This test only ensures that the templates do not
+        # return a 500 error. It doesn't test the logic of the
+        # templates.
+        context = dict(
+            shallow_tested=True,
+            user_profile=user_profile,
+            product_name='testing',
+            form=DummyForm(
+                full_name=get_form_value('John Doe'),
+                terms=get_form_value(True),
+                email=get_form_value('hamlet@zulip.com'),
+            ),
+            current_url=lambda: 'www.zulip.com',
+            referrer=dict(
+                full_name='John Doe',
+                realm=dict(name='zulip.com'),
+            ),
+            uid='uid',
+            token='token',
+            message_count=0,
+            messages=[dict(header='Header')],
+            new_streams=dict(html=''),
+            data=dict(title='Title'),
+        )
+
+        # Just add the templates whose context has a conflict with other
+        # templates' context in `exclude`.
+        exclude = ['analytics/activity.html']
+        templates = [t for t in get_all_templates() if t not in exclude]
+        for template in templates:
+            template = get_template(template)
+            template.render(context)
```

Comment:
I think we want a try/except here so that we can output which template failed to render properly in the test output with a nice message explaining the problem; I think that will save a lot of time for anyone debugging issues this catches